### PR TITLE
Add syslog C extension

### PR DESCRIPTION
**Pull request by bjfish**
Passes 507 tests (4 failures) for the `jt test mri --syslog` command added here.

### CODE REVIEW

File: tool/jt.rb [hunk 1]
```diff
@@ -716,6 +716,10 @@ def test_mri(*args)
     if args.delete('--openssl')
       include_pattern = "#{TRUFFLERUBY_DIR}/test/mri/tests/openssl/test_*.rb"
       exclude_file = "#{TRUFFLERUBY_DIR}/test/mri/openssl.exclude"
+    elsif args.delete('--cext')
```
Comment (eregon) [inline tool/jt.rb]: Maybe we should name this `--syslog`?
And if we have `--cexts` it would include openssl, etc as well?

File: lib/mri/syslog/logger.rb [hunk 2]
```diff
@@ -0,0 +1,209 @@
+# frozen_string_literal: false
```
Comment (eregon) [inline lib/mri/syslog/logger.rb]: This file is from MRI 2.3.3, right?
Can you adapt `tool/update-mri.sh` to include this file and the .c and make sure it works?

Comment (bjfish) [inline lib/mri/syslog/logger.rb]: Yes `2.3.3`. Yes, I'll take a look at the tool.

File: tool/jt.rb (hunk 1 above)
Comment (bjfish) [inline tool/jt.rb]: I've renamed this to `--syslog`. I think a consolidated `--cext` would be a good idea. Could this be done in a separate PR?

File: lib/mri/syslog/logger.rb (hunk 2 above)
Comment (bjfish) [inline lib/mri/syslog/logger.rb]: @eregon I've updated the tool, please review.

File: .rubocop.yml [hunk 3]
```diff
@@ -19,6 +19,7 @@ AllCops:
     - 'tool/**/*'
     - 'truffleruby/target/**/*'
     - 'truffleruby-gem-test-pack-*/**/*'
+    - 'src/main/c/syslog/**'
```
Comment (eregon) [inline .rubocop.yml]: Small thing, but I'm trying to keep those in alphabetic order.
Actually, could you ignore `src/main/c/*/extconf.rb`?
There should not be other .rb files under `src/main/c`.

File: tool/jt.rb (hunk 1 above)
Comment (eregon) [inline tool/jt.rb]: Yes, definitely.

File: .rubocop.yml (hunk 3 above)
Comment (bjfish) [inline .rubocop.yml]: Fixed, i've updated the `.rubocop.yml`.
